tests/Functional/Driver/OCI8/ConnectionTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\OCI8;
+
+use Doctrine\DBAL\DriverManager;
+use Doctrine\DBAL\Exception\DriverException;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+use Throwable;
+
+class ConnectionTest extends FunctionalTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        if (TestUtil::isDriverOneOf('oci8', 'pdo_oci')) {
+            return;
+        }
+
+        self::markTestSkipped('This test requires the oci8 or pdo_oci driver.');
+    }
+
+    public function testPrepareThrowsErrorOnConnectionLost(): void
+    {
+        $this->markConnectionNotReusable();
+        $this->killCurrentSession();
+
+        $this->expectException(DriverException::class);
+
+        $this->connection->executeQuery('SELECT * FROM 1');
+    }
+
+    /**
+     * Kill the current session, by using another connection
+     * Oracle doesn't allow you to terminate the current session, so we use a second connection
+     */
+    private function killCurrentSession(): void
+    {
+        $row = $this->connection->fetchNumeric(
+            <<<'SQL'
+SELECT SID, SERIAL#
+FROM V$SESSION
+WHERE AUDSID = USERENV('SESSIONID')
+SQL,
+        );
+
+        self::assertNotFalse($row);
+        /** @psalm-suppress PossiblyUndefinedArrayOffset */
+        [$sid, $serialNumber] = $row;
+
+        self::assertNotNull($sid, 'SID is missing.');
+        self::assertNotNull($serialNumber, 'Serial number is missing.');
+
+        $params                               = TestUtil::getConnectionParams();
+        $params['driverOptions']['exclusive'] = true;
+        $secondConnection                     = DriverManager::getConnection($params);
+
+        $sessionParam = $this->connection->quote($sid . ', ' . $serialNumber);
+        $secondConnection->executeStatement('ALTER SYSTEM DISCONNECT SESSION ' . $sessionParam . ' IMMEDIATE');
+
+        // Ensure OCI driver is aware of connection state change by executing any statement
+        try {
+            $this->connection->executeStatement('INVALID SQL');
```

Comment:
It doesn't look like this statement is necessary. If I remove it, the test still passes. Could you try removing it or explain why it's necessary?

Reply:
Without it, this error isn't throwing. The same error is being thrown elsewhere and caught by the test.

```php
        if (! is_resource($statement)) {
            throw Error::new($this->connection);
        }
```

This is happening because a few days ago, we changed
```php
$stmt = $this->connection->prepare('SELECT * FROM 1');
```

to
```php
$this->connection->executeQuery('SELECT * FROM 1');
```

Which shouldn't of been changed

This has however exposed that we cant test against PDO_OCI, as PDO_OCI doesnt use `Doctrine\DBAL\Driver\OCI8\Connection`.

I've reverted the line to just using prepare, and made it skip PDO_OCI